### PR TITLE
Import csv-Fix not importing extension in phones

### PR DESCRIPTION
**Pull request by rubofvil**
Overview
----------------------------------------

Fix issue importing the extension in phones by UI.
Related Issue https://lab.civicrm.org/dev/core/-/issues/2713

Before
----------------------------------------
Not works

After
----------------------------------------

- Works the import of the extension phone.
- Is added the selector of the phone_type in the UI.

image

Technical Details
----------------------------------------

Previously is a issue with the keys in the array to related the "phone" with the "phone extension".

### CODE REVIEW

File: CRM/Contact/Import/Form/MapField.php [hunk 1]
```diff
@@ -546,7 +548,7 @@ public function submit($params, $mapperKeys) {
           $locations[$i] = $locationTypes[$selOne];
           $parserParameters['mapperLocType'][$i] = $selOne;
           if ($selTwo && is_numeric($selTwo)) {
-            if ($fldName === 'phone') {
+            if ($fldName === 'phone' || $fldName === 'phone') {
```
Comment (demeritcowboy) [inline CRM/Contact/Import/Form/MapField.php:551]: Based on the rest this should be `phone_ext`

File: CRM/Contact/Import/Form/MapField.php [hunk 2]
```diff
@@ -776,7 +778,7 @@ protected function saveMappingField($mapperKeys, array $saveMapping, string $cTy
         $saveMappingFields->website_type_id = $mapperKeys[$i][1] ?? NULL;
       }
       else {
-        if (CRM_Utils_Array::value('0', $mapperKeys[$i]) == 'phone') {
+        if (CRM_Utils_Array::value('0', $mapperKeys[$i]) == 'phone' || CRM_Utils_Array::value('0', $mapperKeys[$i]) == 'phone_ext') {
```
Comment (demeritcowboy) [inline CRM/Contact/Import/Form/MapField.php:781]: We've been changing CRM_Utils_Array::value into native php as we come across it, to try to get rid of it, like:

`if (($mapperKeys[$i][0] ?? NULL) === 'phone' || ($mapperKeys[$i][0] ?? NULL) === 'phone_ext') {`

File: CRM/Contact/Import/Form/MapField.php (hunk 1 above)
Comment (rubofvil) [reply CRM/Contact/Import/Form/MapField.php]: Fail, thx for the revision 
https://github.com/civicrm/civicrm-core/pull/20995/commits/3aca1423bad7c7dd9d2e0f9687b85823a2b64bb5

File: CRM/Contact/Import/Form/MapField.php (hunk 2 above)
Comment (rubofvil) [reply CRM/Contact/Import/Form/MapField.php]: Replaced by the suggestion
https://github.com/civicrm/civicrm-core/pull/20995/commits/3aca1423bad7c7dd9d2e0f9687b85823a2b64bb5

**Comment by demeritcowboy** [comment]
* General standards
    * [`r-explain`] __PASS__
    * [`r-user`] __PASS__
    * [`r-doc`] __PASS__
    * [`r-run`] __PASS__
      * I tested some simple variations but not too many. I also tested a related contact 
* Developer standards
    * [`r-tech`] __PASS__
    * [`r-code`] __PASS__
      * There's a spot at line 644 that looks like it also needs phone_ext, but I wasn't sure how to get there. When I tested a related contact with a phone extension it worked.
      * Although when I did a related contact it gave me `User deprecated function: Deprecated function CRM_Contact_Import_Parser::formatCommonData, use this is not expected to be reachable now. in CRM_Core_Error::deprecatedFunctionWarning() (line 1044 of .../CRM/Core/Error.php)`, but it looks like that happens before the PR too.
    * [`r-maint`] __?__
      * It would be nice to have a test. Might even be as simple as adding phone_ext into https://github.com/civicrm/civicrm-core/blob/589b3ef9cf1a3656215a1a52ed589fa4592f4490/tests/phpunit/CRM/Contact/Import/Parser/ContactTest.php#L344 and checking the result.
    * [`r-test`] __PASS__

**Comment by demeritcowboy** [comment]
Going to mark merge-ready since the only questions were about a line I don't know how to reach and about adding a unit test.

Jenkins retest this please.

**Comment by demeritcowboy** [comment]
jenkins retest this please - known flaky test time rollover:

```
CRM_Core_BAO_CustomFieldTest::testFileDisplayValueNoDescription
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'              <a href="/index.php?q=civicrm/file&amp;reset=1&amp;id=1&amp;eid=3&amp;fcs=19e65efaae0cb47d0d6b7cc0d606b43611ace0f4d291d54ea01496568631c2a1_1630715045_168" title="test_file.txt">\n
+'              <a href="/index.php?q=civicrm/file&amp;reset=1&amp;id=1&amp;eid=3&amp;fcs=19e65efaae0cb47d0d6b7cc0d606b43611ace0f4d291d54ea01496568631c2a1_1630715046_168" title="test_file.txt">\n
                 <i class="crm-i fa-paperclip" aria-hidden="true"></i>\n
               </a>'

/home/jenkins/bknix-dfl/build/core-20995-7x6xb/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/BAO/CustomFieldTest.php:1004
```

**Pull request merged**